Client/Frontend/Fakespot/FakespotOptInCardViewModel.swift

```diff
@@ -15,6 +15,7 @@ struct FakespotOptInCardViewModel {
     }
 
     private let tabManager: TabManager
+    var themeManager: ThemeManager
```

Comment:
We should keep the `themeManager` in the view controller only as it won't listen for theme updates in view models, and this potentially means using the wrong theme. Another solution could've been to pass down the current `Theme` object to this view model, but in this case it seems to be too complicated to do so.

So I think we can instead revert the changes you just made in `FakespotOptInCardViewModel` and call `mainButton.applyTheme(theme: viewModel.themeManager.currentTheme)` from the `func applyTheme(theme: Theme) {` in `FakespotOptInCardView`.

On top of that change, we'll need to call `view.applyTheme(theme: themeManager.currentTheme)` where the `FakespotOptInCardView` gets created in the `FakeSpotViewController`.

That way the colors should be set properly hopefully, even with live theme changes while that view is showing 🤔 Let me know if it doesn't work out!

Reply:
Yeah, I thought I shouldn't, and I've learned more about view models. I'll go ahead and make the changes you suggested and give them a test.

Reply:
Hi, it looks like the `func applyTheme(theme: Theme)` in `FakespotOptInCardView` is passed by the current theme, I also did a test when I switch system theme, the `FakespotOptInCardView` mainButton will change accordingly, so I guess when `FakespotOptInCardView` gets created in the `FakeSpotViewController` it is already set. So I guess it might be good in this commit ([f8fa055](https://github.com/mozilla-mobile/firefox-ios/pull/16925/commits/f8fa0555966fd4217027bed6182b67291f0c9213))?